### PR TITLE
Add basic component and partial benchmarks

### PR DESCRIPTION
**Pull request by BlakeWilliams**
This adds a basic benchmark script using `benchmark/ips` to compare the
iterations per-second of component renders to partial renders.

This attempts to make the comparison equal by using the `test`
environments Rails application in production mode in addition to going
through a controllers `view_context` to render the partial and the
component.

If anything looks off, or if there's a way to improve the performance of either, please let me know!

Here's what I got running this locally:

```
Warming up --------------------------------------
          component:    23.063k i/100ms
            partial:     2.086k i/100ms
Calculating -------------------------------------
          component:    218.316k (± 8.0%) i/s -      2.191M in  10.105180s
            partial:     19.514k (±11.3%) i/s -    193.998k in  10.074930s

Comparison:
          component::   218316.1 i/s
            partial::    19513.7 i/s - 11.19x  (± 0.00) slower
```

### CODE REVIEW

**Comment by jonspalmer** [comment]
I like it. Simple and to the point.

I'd be curious what the numbers look like for a more complicated view to render.

File: README.md [hunk 1]
```diff
@@ -47,6 +47,12 @@ Traditional Rails views have an implicit interface, making it hard to reason abo
 
 ViewComponents use a standard Ruby initializer that clearly defines what is needed to render, making them easier (and safer) to reuse than partials.
 
+#### Performance
+
+Based on our benchmarks, components are 10x faster than partials. You can find the benchmark script we used to get that number [here](performance/benchmark.rb). If you'd like to run the benchmarks locally, you can run `bundle exec rake benchmark`.
```
Comment (joelhawksley) [inline README.md:52]:
```suggestion
Based on our [benchmarks](performance/benchmark.rb), ViewComponents are ~10x faster than partials. Run `bundle exec rake benchmark` to see for yourself!
```

File: README.md [hunk 2]
```diff
@@ -47,6 +47,12 @@ Traditional Rails views have an implicit interface, making it hard to reason abo
...
+
+The benchmark script is a work-in-progress, so improvements are welcome!
+
```
Comment (joelhawksley) [inline README.md:55]:
```suggestion
```

File: performance/benchmark.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
```
Comment (joelhawksley) [inline performance/benchmark.rb:2]:
```suggestion

# Run `bundle exec rake benchmark` to execute benchmark.
# This is very much a work-in-progress. Please feel free to make/suggest improvements!

```

**Pull request merged**